src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -137,3 +138,66 @@ def _bump_rule_lifetime(project: Project, event_data: Mapping[str, Any]) -> None
             # Only one clustering rule is applied per project
             clusterer_rules.bump_last_used(ClustererNamespace.TRANSACTIONS, project, pattern)
             return
+
+
+def get_span_descriptions(project: Project) -> Iterator[str]:
+    """Return all span descriptions stored for the given project."""
+    client = get_redis_client()
+    redis_key = _get_redis_key(ClustererNamespace.SPANS, project)
+    return client.sscan_iter(redis_key)  # type: ignore
+
+
+def clear_span_descriptions(project: Project) -> None:
+    client = get_redis_client()
+    redis_key = _get_redis_key(ClustererNamespace.SPANS, project)
+    client.delete(redis_key)
+
+
+def record_span_descriptions(
+    project: Project, event_data: Mapping[str, Any], **kwargs: Any
+) -> None:
+    if not features.has("projects:span-metrics-extraction", project):
+        return
+
+    spans = event_data.get("spans", [])
+    for span in spans:
+        if _should_store_span_description(span):
+            description = _get_description_from_span(span)
+            if not description:
+                continue
+            url_path = _get_url_path_from_description(description)
+            safe_execute(_store_span_description, project, url_path)
+
+
+def _should_store_span_description(span: Mapping[str, Any]) -> bool:
+    if not span.get("op", "").startswith("http"):
+        return False
+    data = span.get("data", {})
+    if not data.get("description.scrubbed") and not span.get("description"):
+        return False
+    return True
+
+
+def _get_description_from_span(span: Mapping[str, Any]) -> Optional[str]:
+    data = span.get("data", {})
+    return data.get("description.scrubbed") or span.get("description")
```

Comment:
nit: These two could be a single function
```suggestion
def _get_description_from_span(span: Mapping[str, Any]) -> Optional[str]:
    if not span.get("op", "").startswith("http"):
        return None
    data = span.get("data", {})
    return data.get("description.scrubbed") or span.get("description")


    
```